core/core-impl/src/main/java/org/kontinuity/catapult/core/impl/CatapultImpl.java

```diff
@@ -50,91 +43,86 @@
-
-            if (projectile instanceof ForkProjectile) {
-                ForkProjectile forkProjectile = (ForkProjectile) projectile;
-                /*
-                 * Construct the full URI for the pipeline template file,
-                 * relative to the repository root
-                 */
-                final URI pipelineTemplateUri = UriBuilder.fromUri("https://raw.githubusercontent.com/")
-                        .path(forkProjectile.getSourceGitHubRepo())
-                        .path(forkProjectile.getGitRef())
-                        .path(forkProjectile.getPipelineTemplatePath()).build();
-
-                // Configure the OpenShift project
-                openShiftService.configureProject(createdProject,
-                                                  gitHubRepository.getGitCloneUri(),
-                                                  forkProjectile.getGitRef(),
-                                                  pipelineTemplateUri);
-            } else {
-                openShiftService.configureProject(createdProject, gitHubRepository.getGitCloneUri());
-            }
 
-            final URL webhookUrl = openShiftService.getWebhookUrl(createdProject);
-            if (webhookUrl != null) {
-                try {
-                    webhook = gitHubService.createWebhook(gitHubRepository, webhookUrl, GitHubWebhookEvent.PUSH);
-                } catch (final DuplicateWebhookException dpe) {
-                    // Swallow, it's OK, we've already forked this repo
-                    log.log(Level.INFO, dpe.getMessage());
-                    webhook = ((GitHubServiceSpi) gitHubService)
-                            .getWebhook(gitHubRepository, webhookUrl);
-                }
-            } else {
-                webhook = null;
+        /*
+          TODO Figure how to best handle posilbe DuplicateProjectException, but has to be handled to the user at some intelligent level
+         */
+        final OpenShiftProject createdProject = openShiftService.createProject(projectName);
+
+        ForkProjectile forkProjectile = projectile;
+        /*
+         * Construct the full URI for the pipeline template file,
+         * relative to the repository root
+         */
+        final URI pipelineTemplateUri = UriBuilder.fromUri("https://raw.githubusercontent.com/")
+                .path(forkProjectile.getSourceGitHubRepo())
+                .path(forkProjectile.getGitRef())
+                .path(forkProjectile.getPipelineTemplatePath()).build();
+
+        // Configure the OpenShift project
+        openShiftService.configureProject(createdProject,
+                                          gitHubRepository.getGitCloneUri(),
+                                          forkProjectile.getGitRef(),
+                                          pipelineTemplateUri);
+
+        GitHubWebhook webhook = getGitHubWebhook(gitHubService, gitHubRepository, createdProject);
```

Comment:
this is not code that I introduced, we can update it, but let's do that in a separate PR